### ISSUE TITLE
Geolocation.getCurrentPosition falsely fires errorCallback when setting timeout on iOS

### ISSUE THREAD

**Issue by jdrorrer** [Agontuk/react-native-geolocation-service#291]
**Actual:**
When setting a timeout manually for `Geolocation.getCurrentPosition` on iOS, the errorCallback is called at the end of the timeout no matter what. This makes it difficult to properly handle errors when retrieving a user's location and also causes error monitoring software to get blown up with false negatives.

In our case specifically, we show a list of store locations nearby on app init - if the app could not find their location, we would clear out the user's data and show them an alert stating "Could not find location" since the app is not useable without location data. This would render the app unusable even with the user's location data.

However, this package does cache their location - because the successCallback actually was called the first time around, if the user closes the app then re-opens it, it will have their location stored and show their relevant location-based data.

**Expected:**
When a user's location is found I would expect the successCallback to be called but not the errorCallback. The errorCallback should only be called if the timeout expires without successCallback being called.

**Comment by jdrorrer** [comment]
To fix this, I updated the RNFusedLocation.swift file and used the `patch-package` library with the following diff:

```
diff --git a/node_modules/react-native-geolocation-service/ios/RNFusedLocation.swift b/node_modules/react-native-geolocation-service/ios/RNFusedLocation.swift
index 5538e1e..7aa5bc5 100644
--- a/node_modules/react-native-geolocation-service/ios/RNFusedLocation.swift
+++ b/node_modules/react-native-geolocation-service/ios/RNFusedLocation.swift
@@ -89,6 +89,7 @@ class RNFusedLocation: RCTEventEmitter {
     let distanceFilter = options["distanceFilter"] as? Double ?? kCLDistanceFilterNone
     let maximumAge = options["maximumAge"] as? Double ?? Double.infinity
     let timeout = options["timeout"] as? Double ?? Double.infinity
+    var success: Bool = false
 
     if !lastLocation.isEmpty {
       let elapsedTime = (Date().timeIntervalSince1970 * 1000) - (lastLocation["timestamp"] as! Double)
@@ -96,6 +97,7 @@ class RNFusedLocation: RCTEventEmitter {
       if elapsedTime < maximumAge {
         // Return cached location
         successCallback([lastLocation])
+        success = true
         return
       }
     }
@@ -115,6 +117,7 @@ class RNFusedLocation: RCTEventEmitter {
         target: self,
         selector: #selector(timerFired),
         userInfo: [
+          "success": success,
           "errorCallback": errorCallback,
           "manager": locManager
         ],
@@ -158,6 +161,9 @@ class RNFusedLocation: RCTEventEmitter {
     let data = timer.userInfo as! [String: Any]
     let errorCallback = data["errorCallback"] as! RCTResponseSenderBlock
     let manager = data["manager"] as! CLLocationManager
+    let success = data["success"] as! Bool
+
+    guard success == true else { return }
 
     manager.stopUpdatingLocation()
     manager.delegate = nil
```

**Comment by Agontuk** [comment]
Please try the latest version & see if it solves the issue.

**Comment by jdrorrer** [comment]
@Agontuk this appears to be fixed in the latest version (5.3.0-beta.4 as of today). When I wrote the patch above, I was on version 5.3.0-beta.1 - just a heads up for anyone else coming across this issue.

Thank you @Agontuk!
